### DISCUSSION TITLE
Allow extension of Locks before expiry

### DISCUSSION THREAD

**Discussion by Dynalon** [10x-development/lockmeout.online#5] [Ideas]
Currently, the maximum lock-time is 90days (for locks created with an account). The limit is set in place to minimize database/storage use for a user - especially since some user probably will forget about the lock and never use it again.

In order to allow all users to create "infinite duration" locks, I propose the feature that lock duration time can be extended at some point in time. The initial lock-time remains capped at 90days, but it can be extended for another 90 days shortly before the lock opens. This enforced user interaction ensures only locks are kept in database/storage, where its owners really care about the content.

Some points:
* Emails should be sent out for long-running locks (>14 days) when they are close to open up (maybe 72h before) - this way we can remind the user to extend the locktime before it opens

**Comment by distbit0** [comment]
Would be useful to be able to pay to have infinite duration locks. I'd imagine the cost of storing a few dozen bytes in a database for several years should not be too high :)

**Reply by Dynalon** [reply]
True, the issue "infinite" is a long time. I might think about lifting the 90days limit up to 12 month, at least for non-image locks (those consume quite more storage space)

**Reply by distbit0** [reply]
Yeah personally I'd only be interested in text locks. So at most a few kb or so of text. You can always simply unlock and email the stored values to all users in the event you decide to shutdown the service. 

I think for me a few years would be ideal for peace of mind in case I forget about it and for some reason do not see the email reminders.

**Comment by valimikayilov** [comment]
Even after expiry if we could lock it back again for some time, because we haven’t finished the task at hand, that would be really helpful.

**Reply by Dynalon** [reply]
this is now tracked in #9

**Comment by Dynalon** [comment]
Closing this, as extension of locks is implemented as of today.
